### PR TITLE
 Refactor ves_icall_System_Threading_Monitor_Monitor_pulse and pulse_all.

### PR DESCRIPTION
**Pull request by jaykrell**
These  functions are the same except for if vs. while, so can be one function with a boolean parameter. (and unused func in debugging)

### CODE REVIEW

File: mono/metadata/monitor.c [hunk 1]
```diff
@@ -1292,46 +1292,28 @@ ves_icall_System_Threading_Monitor_Monitor_pulse (MonoObject *obj)
...
+}
+
 void
 ves_icall_System_Threading_Monitor_Monitor_pulse_all (MonoObject *obj)
 {
-	int id;
-	LockWord lw;
-	MonoThreadsSync *mon;
-	
-	LOCK_DEBUG (g_message("%s: (%d) Pulsing all %p", __func__, mono_thread_info_get_small_id (), obj));
```
Comment (jaykrell) [inline mono/metadata/monitor.c]: "all" in the format string is another divergence but redundant with func.

**Comment by jaykrell** [comment]
@monojenkins build failed

File: mono/metadata/monitor.c [hunk 2]
```diff
@@ -1292,46 +1292,29 @@ ves_icall_System_Threading_Monitor_Monitor_pulse (MonoObject *obj)
...
+again:
 	if (mon->wait_list != NULL) {
-		LOCK_DEBUG (g_message ("%s: (%d) signalling and dequeuing handle %p", __func__, mono_thread_info_get_small_id (), mon->wait_list->data));
-	
+		LOCK_DEBUG (g_message ("%s: (%d) signalling and dequeuing handle %p", func, mono_thread_info_get_small_id (), mon->wait_list->data));
+
 		mono_w32event_set (mon->wait_list->data);
 		mon->wait_list = g_slist_remove (mon->wait_list, mon->wait_list->data);
+		if (all)
+			goto again;
```
Comment (lewurm) [inline mono/metadata/monitor.c]: can we use a loop instead of `goto`?

Comment (jaykrell) [reply mono/metadata/monitor.c]: Yes but it isn’t pretty. Something like do while all and not null.

Comment (jaykrell) [reply mono/metadata/monitor.c]: Or while true and break if null or not all.

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build  failed

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build OS X x64

**Comment by jaykrell** [comment]
@monojenkins build Linux x64

**Comment by jaykrell** [comment]
@monojenkins build Linux x64

**Pull request merged**